pkg/odo/cli/service/create.go

```diff
@@ -237,76 +273,99 @@ func (o *ServiceCreateOptions) Validate() (err error) {
 			if err != nil {
 				return err
 			}
-			// var jsonCR map[string]interface{}
-			err = yaml.Unmarshal(fileContents, &o.CustomResourceDefinition)
+
+			err = yaml.Unmarshal(fileContents, &d.OriginalCRD)
 			if err != nil {
 				return err
 			}
 
 			// Check if the operator and the CR exist on cluster
-			o.CustomResource = o.CustomResourceDefinition["kind"].(string)
-			csvs, err := o.KClient.GetClusterServiceVersionList()
+			var csv olm.ClusterServiceVersion
+			o.CustomResource, csv, err = svc.GetCSV(o.KClient, d.OriginalCRD)
 			if err != nil {
 				return err
 			}
 
-			csv, err := doesCRExist(o.CustomResource, csvs)
+			// all is well, let's populate the fields required for creating operator backed service
+			o.group, o.version, o.resource, err = svc.GetGVRFromOperator(csv, o.CustomResource)
 			if err != nil {
-				return fmt.Errorf("Could not find specified service/custom resource: %s\nPlease check the \"kind\" field in the yaml (it's case-sensitive)", o.CustomResource)
+				return err
 			}
 
-			// all is well, let's populate the fields required for creating operator backed service
-			o.group, o.version = groupVersionALMExample(o.CustomResourceDefinition)
-			o.resource = resourceFromCSV(csv, o.CustomResource)
-			o.ServiceName, err = serviceNameFromCRD(o.CustomResourceDefinition, o.ServiceName)
-			return err
-		}
-		if o.CustomResource != "" {
-			// make sure that CSV of the specified ServiceType exists
-			csv, err := o.KClient.GetClusterServiceVersion(o.ServiceType)
+			err = d.validateMetadataInCRD()
 			if err != nil {
-				// error only occurs when OperatorHub is not installed.
-				// k8s does't have it installed by default but OCP does
 				return err
 			}
 
-			var almExamples []map[string]interface{}
-			val, ok := csv.Annotations["alm-examples"]
-			if ok {
-				err = json.Unmarshal([]byte(val), &almExamples)
-				if err != nil {
-					return errors.Wrap(err, "unable to unmarshal alm-examples")
+			if o.ServiceName != "" && !o.DryRun {
+				// First check if service with provided name already exists
+				svcFullName := strings.Join([]string{o.CustomResource, o.ServiceName}, "/")
+				exists, _ := svc.OperatorSvcExists(o.KClient, svcFullName)
```

Comment:
Any reason to ignore the error here?

Reply:
We're checking if the service exists. The function returns `true` if service exists and an error if it fails to find a service with the given name. In `odo service create`, we are concerned with a pre-existing service with the same name. It's not important if it fails to find a service with the given name in `odo service create`. Hence, we're ignoring the error here.

Reply:
We can use kerrors package to differentiate between the error e.g https://github.com/openshift/odo/blob/c89c1ab1d730fb8495f91ce6ea0ed7ff8f784711/pkg/occlient/occlient.go#L3316-L3320

We need to check the error as in case of a network error, it's better to inform the user and stop the execution. Otherwise it might lead to confusion.

Reply:
I have made the change based on our discussion but, tbh, I'm not so sure about how kerrors works. :man_shrugging: